### PR TITLE
feat(report_merger): add Report Merger script

### PR DESCRIPTION
**Pull request by Trinaa**
## Description

- Add the Report Merger script
- Update documentation
  - Restructure documents to follow the SUMMARY.md structure
  - Add the developer setup document
- Remove special characters from CSV report header, since they aren't compatible with BigQuery

- [x] This PR conforms to the [Contribution Guidelines](/CONTRIBUTING)
- [x] AI tools were used in generating this pull request

## Issues

Closes: # [ECTEEN-146](https://mozilla-hub.atlassian.net/browse/ECTEEN-146)

[ECTEEN-146]: https://mozilla-hub.atlassian.net/browse/ECTEEN-146?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: scripts/report_merger/config.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+from configparser import NoSectionError, NoOptionError
+from json import JSONDecodeError
+
+
+from pydantic import BaseModel, ValidationError, Field
+
+from scripts.common.config import BaseConfig, InvalidConfigError, DIRECTORY_PATTERN
+
+
+class ReportsMergerConfig(BaseModel):
```
Comment (pdehaan) [inline scripts/report_merger/config.py:18]: nit: Should this be Report<i><b>s</b></i> plural or Report singular? (considering `report_merger` is singular)

Comment (pdehaan) [reply scripts/report_merger/config.py:18]: Also this below in report_merger.py

```
class ReportMerger:
    """Report Merger."""
```

**Review by pdehaan — APPROVED** [review]

**Review by pdehaan — APPROVED** [review]

File: scripts/report_merger/config.py (hunk 1 above)
Comment (Trinaa) [reply scripts/report_merger/config.py]: oh nice catch! I did a search and found a bunch more places where i made this mistake. They should be fixed now.

**Comment by Trinaa** [comment]
@pdehaan thank you for the thorough review.

**Pull request merged**